segpy/reader.py

```diff
@@ -264,6 +264,11 @@ def _make_reader(fh, encoding, trace_header_format, endian, progress, dimensiona
 
     assert 1 <= dimensionality <= 3
 
+    trace_offset_catalog, trace_length_catalog, cdp_catalog, line_catalog = tuple(
```

Comment:
I anticipate that we'll be able to remove these checks once we're doing sensible construction of all SEG Y elements. For example, since we don't currently try to make sensible/consistent trace headers, I think these catalogs are a bit nonsensical. I'm currently educating myself a bit more on the domain (What's an ensemble? What's CMP? etc.) to make heads-or-tails of it all.

Reply:
In other words: this check is a crutch to let us test *something*.